go/vt/vtctl/workflow/resharder.go

```diff
@@ -331,12 +325,19 @@ func (rs *resharder) startStreams(ctx context.Context) error {
 		// because we've already confirmed that there were no existing workflows
 		// on the shards when we started, and we want to start all of the ones
 		// that we've created on the new shards as we're migrating them.
-		// We use the comment directive to indicate that this is intentional
-		// and OK.
-		query := fmt.Sprintf("update /*vt+ %s */ _vt.vreplication set state='Running' where db_name=%s",
```

Comment:
The current code starts _*all*_  the workflows on the target for the `dbname`. The intention is to restart all previous streams running on the previous shards, for example, for long-running materialize workflows.

The new code starts only streams for the current workflow. As discussed we should update the TestVReplicationBasicWorkflow e2e test to check for running streams at the end of the reshards, so we can catch these issues.

Reply:
Test part: https://github.com/vitessio/vitess/pull/14424/commits/cfddc38102f3a97a5597ddce967326824ed93dcf

Reply:
I've address this here:  https://github.com/vitessio/vitess/pull/14424/commits/9831ee57f9616b60c28a429aa8037122e80a0afe